library/std/src/sys/unix/process/process_common.rs

```diff
@@ -510,16 +511,58 @@ impl ChildStdio {
 }
 
 impl fmt::Debug for Command {
+    // show all attributes but `self.closures` which does not implement `Debug`
+    // and `self.argv` which is not useful for debugging
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if self.program != self.args[0] {
-            write!(f, "[{:?}] ", self.program)?;
-        }
-        write!(f, "{:?}", self.args[0])?;
+        if f.alternate() {
+            let mut debug_command = f.debug_struct("Command");
+            debug_command
+                .field("program", &self.program)
+                .field("args", &self.args)
+                .field("env", &self.env)
+                .field("cwd", &self.cwd)
+                .field("uid", &self.uid)
+                .field("gid", &self.gid)
+                .field("saw_nul", &self.saw_nul)
+                .field("groups", &self.groups);
+
+            if self.stdin.is_some() {
+                debug_command.field("stdin", &self.stdin);
+            }
+            if self.stdout.is_some() {
+                debug_command.field("stdout", &self.stdout);
+            }
+            if self.stderr.is_some() {
+                debug_command.field("stderr", &self.stderr);
+            }
+
+            debug_command.field("pgroup", &self.pgroup);
+
+            #[cfg(target_os = "linux")]
+            {
+                debug_command.field("create_pidfd", &self.create_pidfd);
+            }
 
-        for arg in &self.args[1..] {
-            write!(f, " {:?}", arg)?;
+            debug_command.finish()
+        } else {
+            if let Some(ref cwd) = self.cwd {
+                write!(f, "cd {cwd:?} && ")?;
+            }
+            for (key, value_opt) in self.get_envs() {
+                if let Some(value) = value_opt {
+                    write!(f, "{key:?}={value:?} ")?;
+                }
+            }
```

Comment:
This will not show the env variables that are added when creating the `Command` then removed by `env_remove`. I assume this is ok